ipatests/prci_definitions/prci_checker.py

```diff
@@ -0,0 +1,198 @@
+                f"\nJob name must start with prefix {job_prefix}"
+            )
+        if params.get("requires") != requires:
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "requires" field must'
+                f" be defined as requires: {requires}"
+            )
+        if args.get("build_url") != build_url:
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "build_url" arg field'
+                f" must be defined as build_url: {build_url}"
+            )
+        if (
+            args.get("template") != template
+        ):  # Check template field against build target
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "template" arg field'
+                f" must be defined as template: {template}"
+            )
+        if (
+            copr is not None and args.get("copr") != copr
+        ):  # If build target has a copr repo, check job defines it as well
+            click.echo(
+                ERROR
+                + f"{job_name} in {filename} file."
+                f'\nJob "copr" arg field must be defined as copr: {copr}'
+            )
+        if (
+            update_packages is not None
+            and args.get("update_packages") != update_packages
+        ):
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "update_packages" arg'
+                " field must be defined as update_packages:"
+                f" {update_packages}"
+            )
+        if selinux is not None and args.get("selinux_enforcing") != selinux:
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "selinux_enforcing" '
+                f"arg field must be defined as selinux_enforcing: {selinux}"
+            )
+
+
+@click.command()
+@click.help_option("--help", "-h")
+@click.option(
+    "--file", "-f", help="Specify YAML definition file to be analyzed"
+)
+def main(file):
+    """
+    Checker script for prci definition files.\n
+    This script checks whether jobs in a prci definition file have the correct
+    naming format, requirements, and arguments, which are defined in the
+    specification file.
+
+    Examples of the usage for the tool:\n
+    # Checks all yaml definition files in current dir\n
+    python3 prci_checker.py\n
+    # Checks only specified file\n
+    python3 prci_checker.py -f gating.yaml
+    """
+
+    # Load file with jobs specification format
+    try:
+        jobs_spec = load_yaml(JOBS_SPEC_YAML)
+    except FileNotFoundError:
+        click.echo(
+            ERROR + f"Jobs specification file {JOBS_SPEC_YAML} not found."
+        )
+        raise
+    if file:
+        try:
+            def_suite = load_yaml(file)
+        except FileNotFoundError:
+            click.echo(ERROR + "Specified file was not found.")
+            raise
```

Comment:
Refactored. Now shorter and to the point messages are printed out instead of the stacktrace.

Also added the option to specify the spec-file as a parameter. Although I think it won't be used much, it's nice to have it.